projects/observability/src/shared/dashboard/data/graphql/table/traces/traces-table-data-source.model.ts

```diff
@@ -32,21 +38,32 @@ export class TracesTableDataSourceModel extends TableDataSourceModel {
       requestType: TRACES_GQL_REQUEST,
       traceType: this.traceType,
       properties: request.columns.filter(column => column.visible).map(column => column.specification),
-      limit: request.position.limit * 2, // Prefetch 2 pages
+      limit: request.position.limit, // Prefetch 2 pages
       offset: request.position.startIndex,
       sort: request.sort && {
         direction: request.sort.direction,
         key: request.sort.column.specification
       },
       filters: [...filters, ...this.toGraphQlFilters(request.filters)],
-      timeRange: this.getTimeRangeOrThrow()
+      timeRange: this.getTimeRangeOrThrow(),
+      ignoreTotal: this.ignoreTotal
     };
   }
 
-  protected buildTableResponse(response: TracesResponse): TableDataResponse<TableRow> {
+  protected buildTableResponse(
+    response: TracesResponse,
+    request: TableDataRequest<SpecificationBackedTableColumnDef>
+  ): TableDataResponse<TableRow> {
     return {
       data: response.results,
-      totalCount: response.total
+      // We want to avoid showing real totals for traces table.
+      // Provide the `last` code when results are lesser than the limit (aka, we know there are no more results)
+      // Provide the `unknown` code otherwise
+      totalCount: this.ignoreTotal
+        ? response.results.length < request.position.limit
```

Comment:
Will the result length be ever greater than limit?

Reply:
nope.